2020/day-2-password-philosophy/index.ts

```diff
@@ -0,0 +1,33 @@
+const parseList = (list: string) => {
+  return list
+    .trim()
+    .split('\n')
+    .map(entry => entry.split(': '))
+    .map(([policy, password]) => {
+      const [range, char] = policy.trim().split(' ')
+      const [min, max] = range.split('-').map(Number)
+
+      return { min, max, char, password }
+    })
```

Comment:
This is way more idiomatic than my implementation. It's really easy to read as well, nice work.